test/integration/remote_installer_int_test.go

```diff
@@ -15,23 +17,75 @@ import (
 
 type RemoteInstallIntegrationTestSuite struct {
 	tagsuite.Suite
-	installerExe string
+	remoteInstallerExe string
 }
 
 func (suite *RemoteInstallIntegrationTestSuite) TestInstall() {
 	suite.OnlyRunForTags(tagsuite.RemoteInstaller, tagsuite.Critical)
 	ts := e2e.New(suite.T(), true)
 	defer ts.Close()
 
-	suite.setupTest(ts)
+	tests := []struct {
+		Name    string
+		Version string
+		Channel string
+	}{
+		// Disabled until the target installers support the installpath override env var: DX-1350
+		// {"install-release-latest", "", constants.ReleaseBranch},
+		// {"install-prbranch", "", ""},
+		// {"install-prbranch-with-version", constants.Version, constants.BranchName},
+		{"install-prbranch-and-branch", "", constants.BranchName},
+	}
+
+	for _, tt := range tests {
+		suite.Run(fmt.Sprintf("%s (%s@%s)", tt.Name, tt.Version, tt.Channel), func() {
+			ts := e2e.New(suite.T(), false)
+			defer ts.Close()
+
+			suite.setupTest(ts)
+
+			installPath := filepath.Join(ts.Dirs.Work, "install")
+			stateExePath := filepath.Join(installPath, "bin", constants.StateCmd+exeutils.Extension)
+
+			args := []string{}
+			if tt.Version != "" {
+				args = append(args, "--version", tt.Version)
+			}
+			if tt.Channel != "" {
+				args = append(args, "--channel", tt.Channel)
+			}
 
-	cp := ts.SpawnCmdWithOpts(suite.installerExe, e2e.WithArgs("--channel", constants.ReleaseBranch))
-	cp.Expect("Terms of Service")
-	cp.SendLine("y")
-	cp.Expect("Installing")
-	cp.Expect("Installation complete. Press enter to exit")
-	cp.SendLine("")
-	cp.ExpectExitCode(0)
+			cp := ts.SpawnCmdWithOpts(
+				suite.remoteInstallerExe,
+				e2e.WithArgs(args...),
+				e2e.AppendEnv(constants.InstallPathOverrideEnvVarName+"="+installPath),
+			)
+
+			cp.Expect("Terms of Service")
+			cp.SendLine("y")
+			cp.Expect("Installing")
+			cp.Expect("Installation Complete")
+			cp.Expect("Press ENTER to exit")
+			cp.SendLine("")
+			cp.ExpectExitCode(0)
+
+			suite.Require().FileExists(stateExePath)
+
+			cp = ts.SpawnCmdWithOpts(
+				stateExePath,
+				e2e.WithArgs("--version"),
+				e2e.AppendEnv(constants.InstallPathOverrideEnvVarName+"="+installPath),
```

Comment:
nit: This shouldn't be needed here

Reply:
It's needed to keep the tests from conflicting with one another. Without this, it always installs in a global directory, which given the asynchronous nature of tests can cause all sorts of issues.

Reply:
Oh you meant it's not needed for `--version`; I don't think that's true either though, as we rely on the install path when calling the svc.

Reply:
Ahh okay, makes sense 👍